src/traces/bar/plot.js

```diff
@@ -219,15 +213,20 @@ function appendBarText(gd, bar, calcTrace, i, x0, x1, y0, y1) {
     // compute text position
     var prefix = trace.type === 'waterfall' ? 'waterfall' : 'bar';
     var barmode = fullLayout[prefix + 'mode'];
-    var inStackMode = (barmode === 'stack');
-    var inRelativeMode = (barmode === 'relative');
-    var inStackOrRelativeMode = inStackMode || inRelativeMode;
+    var inStackOrRelativeMode = barmode === 'stack' || barmode === 'relative';
 
     var calcBar = calcTrace[i];
     var isOutmostBar = !inStackOrRelativeMode || calcBar._outmost;
 
-    var barWidth = Math.abs(x1 - x0) - 2 * TEXTPAD;  // padding excluded
-    var barHeight = Math.abs(y1 - y0) - 2 * TEXTPAD;  // padding excluded
+    // padding excluded
+    var barWidth = Math.abs(x1 - x0) - 2 * TEXTPAD;
+    var barHeight = Math.abs(y1 - y0) - 2 * TEXTPAD;
+
+    if(!text || textPosition === 'none' ||
+        (calcBar.isBlank && (textPosition === 'auto' || textPosition === 'inside'))) {
```

Comment:
My solution to the problem noticed in https://github.com/plotly/plotly.js/pull/3164/files#r228981528 (in a previous attempt at fixing this bug) which results in no failing mocks.